lib/tables.dart

```diff
@@ -420,10 +420,25 @@ class Table<T> extends Object with SetStateMixin {
   /// and not necessarily for rows[] since it may be being rendered in reverse.
   /// This way, +1 will always move down the visible table.
   @visibleForTesting
-  void selectByIndex(int newIndex) {
+  void selectByIndex(int newIndex,
+      {bool keepVisible = true, String scrollBehavior = 'smooth'}) {
     final CoreElement row = _rowForIndex[newIndex];
     final T data = rows[_translateRowIndex(newIndex)];
     _select(row?.element, data, newIndex);
+
+    if (keepVisible) {
```

Comment:
I think we probably only want to auto-scroll if the selection is close to the bottom or top of the visible area.